### ISSUE TITLE
Validation Error: img element with empty alt attribute must not have role attribute

### ISSUE THREAD

**Issue by robinson2** [Leaflet/Leaflet#7632]
The HTML code of the tiles produced by Leaflet …

`<img crossorigin="" alt="" role="presentation" src="https://api.maptiler.com/maps/voyager/5/16/10.png?key=…" class="leaflet-tile leaflet-tile-loaded" style="width: 512px; height: 512px; transform: translate3d(886px, -296px, 0px); opacity: 1;">`

… has an error when checked via https://validator.w3.org/nu/ :

"An img element which has an alt attribute whose value is the empty string must not have a role attribute."

In the HTML source code I saw that both were implemented for accessibility reasons:

```
/*
Alt tag is set to empty string to keep screen readers from reading URL and for compliance reasons
http://www.w3.org/TR/WCAG20-TECHS/H67
*/
tile.alt = '';

/*
Set role="presentation" to force screen readers to ignore this
https://www.w3.org/TR/wai-aria/roles#textalternativecomputation
*/
tile.setAttribute('role', 'presentation');
```

On the page https://www.w3.org/WAI/tutorials/images/decorative/ both possibilities are presented and regarding the role attribute the W3C writes: "currently, this feature is not as widely supported as using a null alt attribute."

I would therefore suggest that the role attribute is removed at this point, or that you can choose between both options (alt or role) via configuration.

**Environment**
- Leaflet version: latest 1.7.1 master

**Comment by Falke-Design** [comment]
@Malvoz you are an expert for this things, what do you think?

**Comment by Malvoz** [comment]
Using only one of `alt` or `role` on images results in <q>Image</q> being announced by screen readers and a new error message by the validator, respectively.

If you really wanted to suppress the error message(s) right now you could use `<img ... aria-hidden="true" alt>` without `role`. Although `aria-hidden="true"` is widely supported there may be [compatibility issues](https://www.a11yproject.com/posts/2020-05-13-aria-has-perfect-support/) so `alt` would act as a fallback, in which case <q>Image</q> would be announced by screen readers due to no `role="presentation"`.

I would wait for resolution in https://github.com/validator/validator/issues/1000 before making any changes.

**Comment by Malvoz** [comment]
@timveld 

> on what basis do you claim that setting alt="" on an image results in the screen reader speaking "image"?

I did some testing but I can't remember how I performed the tests and with which technologies unfortunately, bummer. A quick test in https://leafletjs.com/SlavaUkraini/examples/quick-start/example.html - TalkBack announces "this image isn't labelled":

https://user-images.githubusercontent.com/26493779/158205887-5aa91f65-8ff0-4e3d-ac6f-d2f00c629605.mp4

but I'm not sure which image it refers to exactly (note that all the images use an empty `alt`). Either it is the focusable marker (which should be fixed for v1.8.0 due to `role="button"`) or it is one of the tiles, in which case even the combination of `role="presentation"` and empty `alt` does not work (potential TalkBack bug). Maybe `aria-hidden="true"` would? I don't know. Feel free to test it out.

> neither aria-hidden nor role=presentation should be necessary. Such superfluous aria is certainly bad practice ("the first rule of Aria is not to use it") and is an apparent violation of the applicable standards (according to W3C validator).

Yes, that's indeed the first rule of ARIA, but ARIA exists for a reason. If it solves a real world problem then it should be used accordingly.

> So my advice is to simply remove aria-presentation.

But someone added `role="presentation"` to leaflet tiles at some point, presumably it fixed something. Have you tested the effects of removing `role="presentation"`? Keep in mind that Leaflet still supports IE and other legacy technologies in which there may be compatibility issues that do not exist in newer technologies.

> Even if some screen readers did read "image" in that case, such screen readers (accidentally or intentionally) ignore the WCAG guidelines.

Sometimes you have to go further than WCAG techniques to solve a problem, WCAG is not the holy grail of accessibility, they're guidelines. Also note that WCAG is for developers, [HTML AAM](https://www.w3.org/TR/html-aam-1.0/) is for user agents.

**Comment by Malvoz** [comment]
>> So my advice is to simply remove aria-presentation.
>
> But someone added `role="presentation"` to leaflet tiles at some point, presumably it fixed something.

https://github.com/Leaflet/Leaflet/issues/3210#issuecomment-95030940 suggests it was used to make screen readers on iPad ignore images. Though it's impossible to tell if they tested a Leaflet version where tiles already had an empty `alt` (that would make them decorative) or not, as [the bug report](https://github.com/Leaflet/Leaflet/issues/3210) did not include the version(s) tested. The bug report was raised about a year after [empty `alt` was introduced](https://github.com/Leaflet/Leaflet/pull/2509/files) however...

Since it's not clear `role="presentation"` actually solved an issue that an empty `alt` didn't, I'm fine with removing it.

---

An empty `alt` alone should be sufficient in most modern UAs. However, as can be seen in https://github.com/Leaflet/Leaflet/issues/7632#issuecomment-1066980024, TalkBack still announces some image as unlabelled.

If we can agree that there really aren't good use cases for setting descriptive `alt` text for tiles, as described in https://github.com/Leaflet/Leaflet/issues/3210#issuecomment-95030940:

> One could argue that map tiles are more than decorative and do provide information, but really the context is provided by the layer information

we should also consider setting tiles to `aria-hidden="true"`, which seems to fix the TalkBack issue.
